packages/design-system/src/components/RadioGroup/RadioGroup.tsx

```diff
@@ -0,0 +1,60 @@
+import * as RadioGroupPrimitive from '@radix-ui/react-radio-group';
+import { Children, cloneElement } from 'react';
+import { sprinkles } from '../../theme/sprinkles.css';
+import { legendStyle, radioGroupRecipe, RadioGroupRecipeVariant } from './RadioGroup.css';
+
+export type RadioGroupProps<T> = Omit<
+  React.ComponentProps<typeof RadioGroupPrimitive.Root>,
+  'aria-label' | 'onChange'
+> & {
+  legend?: string;
+  onChange: (value: T) => void;
+  value: T;
+} & RadioGroupRecipeVariant;
+
+export function RadioGroup<T extends string = string>({
+  children,
+  legend,
+  onChange,
+  orientation = 'horizontal',
+  required,
+  variant,
+  size,
+  ...props
+}: React.PropsWithChildren<RadioGroupProps<T>>) {
+  const clones = Children.map(children as React.ReactElement[], (child) =>
+    cloneElement(child, { className: radioGroupRecipe({ variant, size }) })
+  );
+  return (
+    <fieldset style={{ margin: 0, padding: 0, borderWidth: 0 }}>
+      <legend className={legendStyle}>
+        {legend} {required && '*'}
```

Comment:
```suggestion
        {legend} {required ? '*' : null}
```
just to understand if we risk to have "0" in page if required false